docs/contribute/core.md

```diff
@@ -0,0 +1,111 @@
+# asdf
+
+`asdf` core contribution guide.
+
+## Initial Setup
+
+Fork `asdf` on GitHub and/or Git clone the default branch:
+
+```shell:no-line-numbers
+# clone your fork
+git clone https://github.com/<GITHUB_USER>/asdf.git
+# or clone asdf
+git clone https://github.com/asdf-vm/asdf.git
+```
+
+The tools for core development are managed with `asdf` in this repo's `.tool-versions`. Add the plugins with:
+
+```shell:no-line-numbers
+asdf plugin add bats https://github.com/timgluz/asdf-bats.git
+asdf plugin add shellcheck https://github.com/luizm/asdf-shellcheck.git
+asdf plugin add shfmt https://github.com/luizm/asdf-shfmt.git
+```
```

Comment:
It might be good to note that these tools don't have to be installed via asdf. And if you are new to asdf and going to be breaking things there is an advantage to not using asdf-installed tools to work on asdf.